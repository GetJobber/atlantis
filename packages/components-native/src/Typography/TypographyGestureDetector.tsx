import type { PropsWithChildren } from "react";
import React from "react";
import { Gesture, GestureDetector } from "react-native-gesture-handler";

/**
 * This is a workaround suggested by react-native-gesture-handler to prevent
 * accidental highlighting of text in Android devices
 * https://github.com/software-mansion/react-native-gesture-handler/issues/1372
 */
export function TypographyGestureDetector(
  props: PropsWithChildren<unknown>,
): JSX.Element {
  return <GestureDetector {...props} gesture={Gesture.Native()} />;
}
