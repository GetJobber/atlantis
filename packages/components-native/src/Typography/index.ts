export { Typography } from "./Typography";
export type {
  FontFamily,
  FontStyle,
  FontWeight,
  BaseWeight,
  DisplayWeight,
  BaseStyle,
  DisplayStyle,
  TextColor,
  TextTransform,
  TextSize,
  TextAlign,
  LineHeight,
  LetterSpacing,
  TextAccessibilityRole,
  TextVariation,
  TypographyProps,
  TruncateLength,
} from "./Typography";
export { typographyStyles } from "./Typography.style";
export { TypographyGestureDetector } from "./TypographyGestureDetector";
