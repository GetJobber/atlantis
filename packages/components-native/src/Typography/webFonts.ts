import type { TextStyle } from "react-native";

export const webFonts: { [index: string]: TextStyle } = {
  baseRegularRegular: {
    fontFamily: "'Inter', Helvetica, Arial, sans-serif",
    fontWeight: "400",
  },

  baseRegularMedium: {
    fontFamily: "'Inter', Helvetica, Arial, sans-serif",
    fontWeight: "500",
  },

  baseRegularSemiBold: {
    fontFamily: "'Inter', Helvetica, Arial, sans-serif",
    fontWeight: "600",
  },

  baseRegularBold: {
    fontFamily: "'Inter', Helvetica, Arial, sans-serif",
    fontWeight: "700",
  },

  baseRegularExtraBold: {
    fontFamily: "'Inter', Helvetica, Arial, sans-serif",
    fontWeight: "800",
  },

  displayRegularBold: {
    fontFamily: "'Jobber Pro', 'Poppins', Helvetica, Arial, sans-serif",
    fontWeight: "700",
  },

  displayRegularExtraBold: {
    fontFamily: "'Jobber Pro', 'Poppins', Helvetica, Arial, sans-serif",
    fontWeight: "800",
  },

  displayRegularBlack: {
    fontFamily: "'Jobber Pro', 'Poppins', Helvetica, Arial, sans-serif",
    fontWeight: "900",
  },
};
