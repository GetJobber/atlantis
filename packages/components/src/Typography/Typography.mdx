---
menu: Components
---

import { Playground, Props } from "docz";
import { ComponentStatus } from "@jobber/docx";
import { Typography } from ".";

# Typography

<ComponentStatus stage="pre" responsive="no" accessible="yes" />

Base component that defines typographic standards across Jobber

```ts
import { Modal } from "@jobber/components/Modal";
```

<Playground>
  <Typography>Just some type here.</Typography>
</Playground>

## Props

<Props of={Typography} />

---

## Element

<Playground>
  <Typography element="h1">Change the element</Typography>
</Playground>

## Font size

<Playground>
  <Typography size="extravagant">Size prop changes the font size</Typography>
</Playground>

## Font weight

<Playground>
  <Typography fontWeight="black">Shout</Typography>
</Playground>

## Text case

<Playground>
  <Typography textCase="uppercase">Shout</Typography>
</Playground>

## Emphasis

<Playground>
  <Typography element="span" emphasisType="highlight">
    Highlight
  </Typography>
</Playground>
