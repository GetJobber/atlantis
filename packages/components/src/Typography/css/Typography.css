@import url("https://fonts.googleapis.com/css?family=Poppins:800,900|Source+Sans+Pro:400,400i,700,700i&display=swap");

:root {
  --typography--fontFamily-normal: "Source Sans Pro", Helvetica, Arial,
    sans-serif;
  --typography--fontFamily-display: "Poppins", Helvetica, Arial, sans-serif;

  --typography--fontSize-extravagant: calc(var(--base-unit) * 3); /* 48 */
  --typography--fontSize-jumbo: calc(var(--base-unit) * 2.25); /* 36 */
  --typography--fontSize-largest: calc(var(--base-unit) * 1.5); /* 24 */
  --typography--fontSize-larger: calc(var(--base-unit) * 1.25); /* 20 */
  --typography--fontSize-large: calc(var(--base-unit) * 1.125); /* 18 */
  --typography--fontSize-base: calc(var(--base-unit) * 1); /* 16 */
  --typography--fontSize-small: calc(var(--base-unit) * 0.875); /* 14 */
  --typography--fontSize-smaller: calc(var(--base-unit) * 0.75); /* 12 */

  --typography--lineHeight-large: 1.34;
  --typography--lineHeight-base: 1.25;
  --typography--lineHeight-tight: 1.2;
  --typography--lineHeight-tighter: 1.143;
  --typography--lineHeight-tightest: 1.12;
  --typography--lineHeight-miniscule: 1.08;
}

.base {
  margin: 0;
  padding: 0;
  color: var(--color-blue);
  font-family: var(--typography--fontFamily-normal);
  font-size: var(--typography--fontSize-base);
  font-weight: 400;
  line-height: var(--typography--lineHeight-base);
}
