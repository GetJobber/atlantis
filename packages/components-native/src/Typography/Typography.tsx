import React from "react";
import type {
  AccessibilityProps,
  StyleProp,
  TextProps,
  TextStyle,
  ViewStyle,
} from "react-native";
import { I18nManager, StyleSheet, Text } from "react-native";
import { TypographyGestureDetector } from "./TypographyGestureDetector";
import { useTypographyStyles } from "./Typography.style";
import { capitalize } from "../utils/intl";
import { useAtlantisTheme } from "../AtlantisThemeContext";

export interface TypographyProps<T extends FontFamily> {
  /**
   * Text capitalization
   */
  readonly transform?: TextTransform;

  /**
   * Color of text
   */
  readonly color?: TextColor;

  /**
   * Alignment of text
   */
  readonly align?: TextAlign;

  /**
   * Lets the user select text, to use the native copy and paste functionality.
   * WARNING: if true, this prevents ellipsis from being shown on Android.
   * @default true
   */
  readonly selectable?: boolean;

  /**
   * Font size
   */
  readonly size?: TextSize;

  /**
   * Text to display
   */
  readonly children?: string;

  /**
   * The maximum amount of lines the text can occupy before being truncated with "...".
   * Uses predefined string values that correspond to a doubling scale for the amount of lines.
   * WARNING: if `selectable` is true, Android will not show an ellipsis.
   */
  readonly maxLines?: TruncateLength;

  /**
   * Allow text to be resized based on user's device display scale
   */
  readonly allowFontScaling?: boolean;

  /**
   * Set the maximum font the text can go to size when the user scales their
   * device font size
   */
  readonly maxFontScaleSize?: number;

  /**
   * Determines whether text should be scaled down to fit based on maxLines prop
   */
  readonly adjustsFontSizeToFit?: boolean;

  /**
   * Line Height
   */
  readonly lineHeight?: LineHeight;

  /**
   * Spacing between letters
   */
  readonly letterSpacing?: LetterSpacing;

  /**
   * Font Family
   */
  readonly fontFamily?: T;

  /**
   * Font style
   */
  readonly fontStyle?: T extends "base" ? BaseStyle : DisplayStyle;

  /**
   * Underline style to use for the text. The non-solid style is only supported
   * on iOS, as per React Native's Text component's limitations.
   * https://reactnative.dev/docs/text-style-props#textdecorationstyle-ios
   */
  readonly underline?: "solid" | "double" | "dotted" | "dashed" | undefined;

  /**
   * Font weight
   */
  readonly fontWeight?: T extends "base" ? BaseWeight : DisplayWeight;

  /**
   * Reverse theme for better display on dark background
   */
  readonly reverseTheme?: boolean;

  /**
   * Accessibility role describing the context of the text
   */
  readonly accessibilityRole?: TextAccessibilityRole;

  /**
   * This will make the text inaccessible to the screen reader.
   * This should be avoided unless there is a good reason.
   * For example this is used in InputText to make it so the label isn't
   * selectable because it is already read from the accessibilityLabel
   * of the TextInput
   */
  readonly hideFromScreenReader?: boolean;

  /**
   * Have text styled with strike through
   */
  readonly strikeThrough?: boolean;

  readonly UNSAFE_style?: TypographyUnsafeStyle;

  /**
   * Callback behaves differently on iOS and Android.
   * On iOS, it is called when the text is laid out.
   * On Android, it is called before the text is laid out.
   * @see https://reactnative.dev/docs/text#ontextlayout
   */
  readonly onTextLayout?: OnTextLayoutEvent;
}

const maxNumberOfLines = {
  single: 1,
  small: 2,
  base: 4,
  large: 8,
  extraLarge: 16,
  unlimited: undefined,
};

export interface TypographyUnsafeStyle {
  textStyle?: StyleProp<TextStyle>;
}

export const Typography = React.memo(InternalTypography);

// eslint-disable-next-line max-statements
function InternalTypography<T extends FontFamily = "base">({
  fontFamily,
  fontStyle,
  fontWeight,
  transform,
  color,
  align,
  size = "default",
  children,
  maxLines = "unlimited",
  allowFontScaling = true,
  maxFontScaleSize,
  adjustsFontSizeToFit = false,
  lineHeight,
  letterSpacing = "base",
  reverseTheme = false,
  hideFromScreenReader = false,
  accessibilityRole = "text",
  strikeThrough = false,
  underline,
  UNSAFE_style,
  selectable = true,
  onTextLayout,
}: TypographyProps<T>): React.JSX.Element {
  const styles = useTypographyStyles();
  const sizeAndHeight = getSizeAndHeightStyle(size, styles, lineHeight);
  const style: StyleProp<ViewStyle>[] = [
    getFontStyle(fontFamily, fontStyle, fontWeight, styles),
    getColorStyle(styles, color, reverseTheme),
    getAlignStyle(styles, align),
    sizeAndHeight,
    getLetterSpacingStyle(letterSpacing, styles),
  ];

  if (strikeThrough) {
    style.push(styles.strikeThrough);
  }

  if (fontStyle === "italic") {
    style.push(styles.italic);
  }

  if (underline) {
    const underlineTextStyle: TextStyle = { textDecorationStyle: underline };
    style.push(underlineTextStyle, styles.underline);
  }

  if (UNSAFE_style?.textStyle) {
    style.push(UNSAFE_style.textStyle);
  }

  const numberOfLinesForNativeText = maxNumberOfLines[maxLines];

  const text = getTransformedText(children, transform);
  const accessibilityProps: AccessibilityProps = hideFromScreenReader
    ? {
        accessibilityRole: "none",
        accessible: false,
        importantForAccessibility: "no-hide-descendants",
      }
    : { accessibilityRole };

  const { tokens } = useAtlantisTheme();

  const textComponent = (
    <Text
      {...{
        allowFontScaling,
        adjustsFontSizeToFit,
        style,
        numberOfLines: numberOfLinesForNativeText,
      }}
      {...accessibilityProps}
      maxFontSizeMultiplier={getScaleMultiplier(
        maxFontScaleSize,
        sizeAndHeight.fontSize,
      )}
      selectable={selectable}
      selectionColor={tokens["color-brand--highlight"]}
      onTextLayout={onTextLayout}
    >
      {text}
    </Text>
  );

  // If text is not selectable, there's no need for TypographyGestureDetector
  // since it only prevents accidental highlighting of selectable text
  if (!selectable) {
    return textComponent;
  }

  return <TypographyGestureDetector>{textComponent}</TypographyGestureDetector>;
}

function getScaleMultiplier(maxFontScaleSize = 0, size = 1) {
  if (maxFontScaleSize === 0) return undefined;

  return maxFontScaleSize / size;
}

function getFontStyle(
  fontFamily: FontFamily = "base",
  fontStyle: FontStyle = "regular",
  fontWeight: FontWeight = "regular",
  styles: ReturnType<typeof useTypographyStyles>,
) {
  const defaultBaseFontStyling = styles.baseRegularRegular;
  const defaultDisplayFontStyling = styles.displayRegularBold;
  const styleKey = `${fontFamily}${capitalize(fontStyle)}${capitalize(
    fontWeight,
  )}`;
  const fontStyling = styles[styleKey];

  if (fontStyling) {
    return fontStyling;
  } else {
    return fontFamily === "display"
      ? defaultDisplayFontStyling
      : defaultBaseFontStyling;
  }
}

function getTransformedText(text?: string, transform?: TextTransform) {
  switch (transform) {
    case "lowercase":
      return text?.toLocaleLowerCase();
    case "uppercase":
      return text?.toLocaleUpperCase();
    case "capitalize":
      return capitalize(text || "");
    default:
      return text;
  }
}

function getColorStyle(
  styles: ReturnType<typeof useTypographyStyles>,
  color?: TextColor,
  reverseTheme?: boolean,
) {
  if (color === "default" || !color) {
    return styles.greyBlue;
  }
  const colorStyleKey = `${color}${reverseTheme ? "Reverse" : ""}`;

  return styles[`${colorStyleKey}`];
}

function getAlignStyle(
  styles: ReturnType<typeof useTypographyStyles>,
  alignStyle: TextAlign = I18nManager.isRTL ? "end" : "start",
) {
  return styles[`${alignStyle}Align`];
}

function getSizeAndHeightStyle(
  textSize: TextSize,
  styles: ReturnType<typeof useTypographyStyles>,
  lineHeightOverwrite?: LineHeight,
) {
  const fontSize = styles[`${textSize}Size`];

  if (lineHeightOverwrite) {
    const lineHeight = styles[`${lineHeightOverwrite}LineHeight`];

    return StyleSheet.flatten([fontSize, lineHeight]);
  }

  return fontSize;
}

function getLetterSpacingStyle(
  letterSpacing: LetterSpacing,
  styles: ReturnType<typeof useTypographyStyles>,
) {
  return styles[`${letterSpacing}LetterSpacing`];
}

export type FontFamily = "base" | "display";
export type FontStyle = "regular" | "italic";
export type FontWeight =
  | "regular"
  | "medium"
  | "bold"
  | "semiBold"
  | "extraBold"
  | "black";

export type BaseWeight = Extract<
  FontWeight,
  "regular" | "medium" | "semiBold" | "bold" | "extraBold"
>;

export type DisplayWeight = Extract<
  FontWeight,
  "semiBold" | "bold" | "extraBold" | "black"
>;

export type BaseStyle = FontStyle;
export type DisplayStyle = Extract<FontStyle, "regular">;

export type TextColor =
  | TextVariation
  // Base colors for backwards compatibility
  | "default"
  | "blueDark"
  | "white"
  | "green"
  | "greenDark"
  | "grey"
  | "greyDark"
  | "greyBlue"
  | "greyBlueDark"
  | "lightBlue"
  | "lightBlueDark"
  | "red"
  | "redDark"
  | "yellow"
  | "yellowDark"
  | "yellowGreenDark"
  | "orangeDark"
  | "navyDark"
  | "limeDark"
  | "purpleDark"
  | "pinkDark"
  | "tealDark"
  | "indigoDark"
  | "navy"

  // Typography
  | "heading"
  | "text"
  | "textSecondary"
  | "textReverse"
  | "textReverseSecondary"

  // Statuses
  | "inactive"
  | "inactiveSurface"
  | "inactiveOnSurface"
  | "critical"
  | "criticalSurface"
  | "criticalOnSurface"
  | "warning"
  | "warningSurface"
  | "warningOnSurface"
  | "informative"
  | "informativeSurface"
  | "informativeOnSurface"
  | "successSurface"
  | "successOnSurface"

  // Interactions
  | "interactiveHover"
  | "interactiveSubtleHover"
  | "destructiveHover"
  | "disabledSecondary"

  // Workflow
  | "request"
  | "requestSurface"
  | "requestOnSurface"
  | "quote"
  | "quoteSurface"
  | "quoteOnSurface"
  | "job"
  | "jobSurface"
  | "jobOnSurface"
  | "visit"
  | "visitSurface"
  | "visitOnSurface"
  | "task"
  | "taskSurface"
  | "taskOnSurface"
  | "event"
  | "eventSurface"
  | "eventOnSurface"
  | "invoice"
  | "invoiceSurface"
  | "invoiceOnSurface"
  | "payments"
  | "paymentsSurface"
  | "paymentsOnSurface"
  | "client"

  // Deprecated
  | "blue" // Use "heading" instead
  | "learning" // Use "informative" instead
  | "subtle" // Use "interactiveSubtle" instead
  | "onPrimary"; // Use "subtle" instead

// A subset of colors that can be used with the Text component
export type TextVariation =
  | "text"
  | "textSecondary"
  | "interactive"
  | "interactiveSubtle"
  | "successOnSurface"
  | "critical"
  | "warning"
  | "informativeOnSurface"
  | "disabled"
  | "destructive"

  // Deprecated
  | "base" // Use "text" instead
  | "success" // Use "successOnSurface" instead (except for when reverseTheme=true)
  | "subdued" // Use "textSecondary" instead
  | "error" // Use "critical" instead
  | "warn" // Use "warningOnSurface" instead
  | "info"; // Use "informativeOnSurface" instead

export type TextTransform = "uppercase" | "lowercase" | "capitalize" | "none";

export type TextSize =
  | "smallest"
  | "smaller"
  | "small"
  | "default"
  | "large"
  | "larger"
  | "largest"
  | "jumbo"
  | "extravagant";

export type TextAlign = "start" | "end" | "center" | "justify";

export type LineHeight =
  | "extravagant"
  | "jumbo"
  | "largest"
  | "larger"
  | "large"
  | "base"
  | "tight";

export type LetterSpacing = "base" | "loose";

export type TextAccessibilityRole = "text" | "header";

export type TruncateLength =
  | "single"
  | "small"
  | "base"
  | "large"
  | "extraLarge"
  | "unlimited";

export type OnTextLayoutEvent = Exclude<TextProps["onTextLayout"], undefined>;
