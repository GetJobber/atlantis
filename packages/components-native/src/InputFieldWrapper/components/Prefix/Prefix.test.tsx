import React from "react";
import { render } from "@testing-library/react-native";
import { TextStyle } from "react-native";
import {
  PrefixIcon,
  PrefixIconProps,
  PrefixLabel,
  PrefixLabelProps,
} from "./Prefix";
import { typographyStyles } from "../../../Typography";

const mockLabel = "$";

function setupLabel({
  disabled = false,
  focused = false,
  hasMiniLabel = false,
  inputInvalid = false,
  label = mockLabel,
  styleOverride,
}: Partial<PrefixLabelProps>) {
  return render(
    <PrefixLabel
      disabled={disabled}
      focused={focused}
      hasMiniLabel={hasMiniLabel}
      inputInvalid={inputInvalid}
      label={label}
      styleOverride={styleOverride}
    />,
  );
}

function setupIcon({
  disabled = false,
  focused = false,
  hasMiniLabel = false,
  inputInvalid = false,
  icon = "invoice",
}: Partial<PrefixIconProps>) {
  return render(
    <PrefixIcon
      disabled={disabled}
      focused={focused}
      hasMiniLabel={hasMiniLabel}
      inputInvalid={inputInvalid}
      icon={icon}
    />,
  );
}

describe("Prefix", () => {
  it("renders a prefix label when specified", () => {
    const { getByText } = setupLabel({});
    expect(getByText(mockLabel)).toBeDefined();
  });

  it("renders a prefix icon when specified", () => {
    const { getByTestId } = setupIcon({});
    expect(getByTestId("invoice")).toBeDefined();
  });

  describe("updates the styles when focused", () => {
    it("for the label", () => {
      const tree = setupLabel({
        focused: true,
      }).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("for the icon", () => {
      const tree = setupIcon({
        focused: true,
      }).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("updates the styles when input is invalid", () => {
    it("for the label", () => {
      const tree = setupLabel({
        inputInvalid: true,
      }).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("for the icon", () => {
      const tree = setupIcon({
        inputInvalid: true,
      }).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it("updates the position of the label when a value is entered", () => {
    const tree = setupLabel({
      hasMiniLabel: true,
    }).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("when disabled", () => {
    it("updates the label", () => {
      const tree = setupLabel({
        disabled: true,
      }).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("updates the icon", () => {
      const tree = setupIcon({
        disabled: true,
      }).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("Custom Label Style", () => {
    it("uses default styling when style override not used", () => {
      const { getByText } = setupLabel({});
      const prefix = getByText(mockLabel);
      const expectedStyle = [
        typographyStyles.baseRegularRegular,
        typographyStyles.base,
        typographyStyles.startAlign,
        typographyStyles.defaultSize,
        typographyStyles.baseLetterSpacing,
      ];
      expect(prefix.props.style).toEqual(expectedStyle);
    });

    it("uses style override for label when provided", () => {
      const styleOverride = {
        fontSize: 50,
        color: "purple",
      };
      const { getByText } = setupLabel({ styleOverride });
      const prefix = getByText(mockLabel);
      const flattenedStyle = prefix.props.style.reduce(
        (style: TextStyle, additionalStyles: TextStyle) => ({
          ...style,
          ...additionalStyles,
        }),
        {},
      );

      const expectedStyle = {
        ...styleOverride,
        ...typographyStyles.baseRegularRegular,
        lineHeight: typographyStyles.defaultSize.lineHeight,
      };

      expect(flattenedStyle).toEqual(expectedStyle);
    });
  });
});
