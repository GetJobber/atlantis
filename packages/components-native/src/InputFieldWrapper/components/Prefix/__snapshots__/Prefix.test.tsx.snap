// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prefix updates the position of the label when a value is entered 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      false,
      false,
    ]
  }
>
  <View
    style={
      [
        {
          "justifyContent": "center",
          "paddingRight": 1,
          "paddingTop": 1,
        },
        {
          "bottom": 0,
          "left": 0,
          "paddingTop": 0,
          "right": 0,
          "top": 7,
        },
      ]
    }
  >
    <Text
      accessibilityRole="text"
      adjustsFontSizeToFit={false}
      allowFontScaling={true}
      collapsable={false}
      maxFontSizeMultiplier={3.125}
      selectable={true}
      selectionColor="rgb(160, 215, 42)"
      style={
        [
          {
            "fontFamily": "inter-regular",
          },
          {
            "color": "rgb(66, 78, 86)",
          },
          {
            "textAlign": "left",
          },
          {
            "fontSize": 16,
            "lineHeight": 20,
          },
          {
            "letterSpacing": 0,
          },
        ]
      }
    >
      $
    </Text>
  </View>
</View>
`;

exports[`Prefix updates the styles when focused for the icon 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "borderColor": "rgb(125, 176, 14)",
      },
      false,
    ]
  }
>
  <View
    style={
      {
        "justifyContent": "center",
        "paddingRight": 8,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "display": "flex",
            "fill": "rgb(136, 112, 196)",
            "height": 24,
            "verticalAlign": "middle",
            "width": 24,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      testID="invoice"
      vbHeight={24}
      vbWidth={24}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4287131844,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M6 2a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V7.414A2 2 0 0 0 19.414 6L16 2.586A2 2 0 0 0 14.586 2H6Zm0 18V4h8.586L18 7.414V20H6Zm7.293-10.293C12.626 9.04 12.003 9 11.977 9 11.45 9 11 9.45 11 10c0 .646.128.752 1.243 1.03C13.274 11.288 15 11.72 15 14a2.995 2.995 0 0 1-2 2.814v.184a1 1 0 0 1-2 0v-.2a4.502 4.502 0 0 1-1.707-1.09 1 1 0 1 1 1.414-1.415c.667.667 1.291.707 1.317.707.528 0 .976-.448.976-1 0-.646-.127-.75-1.242-1.03C10.728 12.713 9 12.28 9 10a2.996 2.996 0 0 1 2-2.816V7a1 1 0 0 1 2 0v.2a4.49 4.49 0 0 1 1.707 1.093 1 1 0 0 1-1.414 1.414Z"
          fill={
            {
              "payload": 4284840068,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Prefix updates the styles when focused for the label 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "borderColor": "rgb(125, 176, 14)",
      },
      false,
    ]
  }
>
  <View
    style={
      [
        {
          "justifyContent": "center",
          "paddingRight": 1,
          "paddingTop": 1,
        },
        false,
      ]
    }
  >
    <Text
      accessibilityRole="text"
      adjustsFontSizeToFit={false}
      allowFontScaling={true}
      collapsable={false}
      maxFontSizeMultiplier={3.125}
      selectable={true}
      selectionColor="rgb(160, 215, 42)"
      style={
        [
          {
            "fontFamily": "inter-regular",
          },
          {
            "color": "rgb(66, 78, 86)",
          },
          {
            "textAlign": "left",
          },
          {
            "fontSize": 16,
            "lineHeight": 20,
          },
          {
            "letterSpacing": 0,
          },
        ]
      }
    >
      $
    </Text>
  </View>
</View>
`;

exports[`Prefix updates the styles when input is invalid for the icon 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      false,
      {
        "borderColor": "rgb(201, 66, 33)",
      },
    ]
  }
>
  <View
    style={
      {
        "justifyContent": "center",
        "paddingRight": 8,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "display": "flex",
            "fill": "rgb(136, 112, 196)",
            "height": 24,
            "verticalAlign": "middle",
            "width": 24,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      testID="invoice"
      vbHeight={24}
      vbWidth={24}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4287131844,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M6 2a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V7.414A2 2 0 0 0 19.414 6L16 2.586A2 2 0 0 0 14.586 2H6Zm0 18V4h8.586L18 7.414V20H6Zm7.293-10.293C12.626 9.04 12.003 9 11.977 9 11.45 9 11 9.45 11 10c0 .646.128.752 1.243 1.03C13.274 11.288 15 11.72 15 14a2.995 2.995 0 0 1-2 2.814v.184a1 1 0 0 1-2 0v-.2a4.502 4.502 0 0 1-1.707-1.09 1 1 0 1 1 1.414-1.415c.667.667 1.291.707 1.317.707.528 0 .976-.448.976-1 0-.646-.127-.75-1.242-1.03C10.728 12.713 9 12.28 9 10a2.996 2.996 0 0 1 2-2.816V7a1 1 0 0 1 2 0v.2a4.49 4.49 0 0 1 1.707 1.093 1 1 0 0 1-1.414 1.414Z"
          fill={
            {
              "payload": 4284840068,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Prefix updates the styles when input is invalid for the label 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      false,
      {
        "borderColor": "rgb(201, 66, 33)",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "justifyContent": "center",
          "paddingRight": 1,
          "paddingTop": 1,
        },
        false,
      ]
    }
  >
    <Text
      accessibilityRole="text"
      adjustsFontSizeToFit={false}
      allowFontScaling={true}
      collapsable={false}
      maxFontSizeMultiplier={3.125}
      selectable={true}
      selectionColor="rgb(160, 215, 42)"
      style={
        [
          {
            "fontFamily": "inter-regular",
          },
          {
            "color": "rgb(66, 78, 86)",
          },
          {
            "textAlign": "left",
          },
          {
            "fontSize": 16,
            "lineHeight": 20,
          },
          {
            "letterSpacing": 0,
          },
        ]
      }
    >
      $
    </Text>
  </View>
</View>
`;

exports[`Prefix when disabled updates the icon 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      false,
      false,
    ]
  }
>
  <View
    style={
      {
        "justifyContent": "center",
        "paddingRight": 8,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "display": "flex",
            "fill": "rgb(136, 112, 196)",
            "height": 24,
            "verticalAlign": "middle",
            "width": 24,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      testID="invoice"
      vbHeight={24}
      vbWidth={24}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4287131844,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M6 2a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V7.414A2 2 0 0 0 19.414 6L16 2.586A2 2 0 0 0 14.586 2H6Zm0 18V4h8.586L18 7.414V20H6Zm7.293-10.293C12.626 9.04 12.003 9 11.977 9 11.45 9 11 9.45 11 10c0 .646.128.752 1.243 1.03C13.274 11.288 15 11.72 15 14a2.995 2.995 0 0 1-2 2.814v.184a1 1 0 0 1-2 0v-.2a4.502 4.502 0 0 1-1.707-1.09 1 1 0 1 1 1.414-1.415c.667.667 1.291.707 1.317.707.528 0 .976-.448.976-1 0-.646-.127-.75-1.242-1.03C10.728 12.713 9 12.28 9 10a2.996 2.996 0 0 1 2-2.816V7a1 1 0 0 1 2 0v.2a4.49 4.49 0 0 1 1.707 1.093 1 1 0 0 1-1.414 1.414Z"
          fill={
            {
              "payload": 4290098613,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Prefix when disabled updates the label 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      false,
      false,
    ]
  }
>
  <View
    style={
      [
        {
          "justifyContent": "center",
          "paddingRight": 1,
          "paddingTop": 1,
        },
        false,
      ]
    }
  >
    <Text
      accessibilityRole="text"
      adjustsFontSizeToFit={false}
      allowFontScaling={true}
      collapsable={false}
      maxFontSizeMultiplier={3.125}
      selectable={true}
      selectionColor="rgb(160, 215, 42)"
      style={
        [
          {
            "fontFamily": "inter-regular",
          },
          {
            "color": "rgb(181, 181, 181)",
          },
          {
            "textAlign": "left",
          },
          {
            "fontSize": 16,
            "lineHeight": 20,
          },
          {
            "letterSpacing": 0,
          },
        ]
      }
    >
      $
    </Text>
  </View>
</View>
`;
